src/extension.ts

```diff
@@ -135,7 +139,7 @@ async function openSarifFileIfNotOpen(sarifFile: vscode.Uri): Promise<void> {
  * @param diagnosticCollection The diagnostic collection to add the results too.
  * @param options Options controlling the upgrade prompt if a SARIF file is at an earlier schema version.
  */
-async function openSarifFile(sarifFile: vscode.Uri,  logReader: LogReader, diagnosticCollection: SVDiagnosticCollection, options: OpenSarifFileOptions): Promise<void> {
+export async function openSarifFile(sarifFile: vscode.Uri, logReader: LogReader, diagnosticCollection: SVDiagnosticCollection, options: OpenSarifFileOptions): Promise<void> {
     if (!sarifFile.isSarifFile()) {
         return;
```

Comment:
Per my other comment, this one isn't a `Promise.resolve()`?

Reply:
This is my preference for sure. The other one was because I wanted to make it possible for the API to be async.

I can\should switch the API to thenable, which allows for either I believe.

---
In reply to: [409124640](https://github.com/microsoft/sarif-vscode-extension/pull/248#discussion_r409124640) [](ancestors = 409124640)

Reply:
I added some comments.

---
In reply to: [409145629](https://github.com/microsoft/sarif-vscode-extension/pull/248#discussion_r409145629) [](ancestors = 409145629,409124640)

Reply:
it returns a "void" since the function is marked as "async" - the return value should not be a Promise, but just the type encapsulated by the Promise.

---
In reply to: [409147124](https://github.com/microsoft/sarif-vscode-extension/pull/248#discussion_r409147124) [](ancestors = 409147124,409145629,409124640)